lighthouse-core/computed/metrics/cumulative-long-queuing-delay.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('../computed-artifact.js');
+const ComputedMetric = require('./metric.js');
+const LHError = require('../../lib/lh-error.js');
+const TracingProcessor = require('../../lib/traces/tracing-processor.js');
+const LanternCumulativeLongQueuingDelay = require('./lantern-cumulative-long-queuing-delay.js');
+const TimetoInteractive = require('./interactive.js');
+
+/**
+ * @fileoverview This audit determines the largest 90 percentile EQT value of all 5s windows between
+ *    FMP and the end of the trace.
+ * @see https://docs.google.com/document/d/1b9slyaB9yho91YTOkAQfpCdULFkZM9LqsipcX3t7He8/preview
+ */
+class CumulativeLongQueuingDelay extends ComputedMetric {
+  /**
+   * @return {number}
+   */
+  static get LONG_QUEUING_DELAY_THRESHOLD() {
+    return 50;
+  }
+  /**
+   * @param {Array<{start: number, end: number, duration: number}>} topLevelEvents
+   * @param {number} fcpTimeInMs
+   * @param {number} interactiveTimeMs
+   * @return {number}
+   */
+  static calculateSumOfLongQueuingDelay(topLevelEvents, fcpTimeInMs, interactiveTimeMs) {
+    const threshold = CumulativeLongQueuingDelay.LONG_QUEUING_DELAY_THRESHOLD;
+    const longQueuingDelayRegions = [];
+    // [              250ms Task                   ]
+    // |  Long Queuing Delay Region  |   Last 50ms |
+    //           200 ms
+    for (const event of topLevelEvents) {
+      if (event.duration < threshold) continue;
+      longQueuingDelayRegions.push({
+        start: event.start,
+        end: event.end - threshold,
+        duration: event.duration - threshold,
+      });
+    }
+
+    let sumLongQueuingDelay = 0;
+    for (const region of longQueuingDelayRegions) {
+      if (region.end < fcpTimeInMs) continue;
```

Comment:
a few comments just explaining what's going on would be great, I've found myself slowly trying to get our metric implementations more space shuttle style 


examples (feel free to word however you wish though for what makes sense with metric defn)

```suggestion
      // If the entire region is before FCP, it doesn't really affect the user, skip it.
      if (region.end < fcpTimeInMs) continue;
      // If the entire region is after TTI, it didn't have a large impact, skip it.
      if (region.start > interactiveTimeMs) continue;
```